content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1433 @@
+[`library/6_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/6_helper.js).
+[`library/7_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/7_helper.js).
+[`view/7_preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/7_preload.js).
+[`view/7_send-xrp.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/7_send-xrp.html).
+[`view/7_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/7_renderer.js).
+
+Up until now we have enabled our app to query and display data from the XRPL. Now it's time to actively participate in the ledger by enabling our application to send transactions. For now, we can stick to sending direct XRP payments because there are more complexities involved in sending issued tokens. After finishing this step the application should look like this:
+
+![Screenshot: Step 7, send xrp dialog](img/javascript-wallet-7.png)
+
+1. Create the file `library/7_helpers.js` and add the following contents:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/library/7_helpers.js", language="js") }}
+
+2. Add the new function to the import section in `index.js`:
+
+```javascript
+const { initialize, subscribe, saveSaltedSeed, loadSaltedSeed } = require('./library/5_helpers')
+const { sendXrp } = require('./library/7_helpers')
+```
+
+3. Still in `index.js`, add an event listener handling the `send-xrp-event` from the frontend dialog:
+```javascript
+        await initialize(client, wallet, appWindow)
+        // Step 7 code additions - start
+        ipcMain.on('send-xrp-action', (event, paymentData) => {
+          sendXrp(paymentData, client, wallet).then((result) => {
+            appWindow.webContents.send('send-xrp-transaction-finish', result)
+          })
+        })
+        // Step 7 code additions - start
+    })
+```
+
+4. Modify `view/preload.js` by adding two new functions:
+```javascript
+    onClickSendXrp: (paymentData) => {
+      ipcRenderer.send('send-xrp-action', paymentData)
+    },
+    onSendXrpTransactionFinish: (callback) => {
+      ipcRenderer.on('send-xrp-transaction-finish', callback)
+    }
+```
+
+5. In `view/template.html`, add a button to toggle the modal dialog housing the "Send XRP" logic:
+```html
+    <div class="header border-bottom">
+        <h3>
+            Build a XRPL Wallet
+            <small class="text-muted">- Part 7/8</small>
+        </h3>
+        <button type="button" class="btn btn-primary" id="send-xrp-modal-button">
+            Send XRP
+        </button>
+    </div>
+```
+
+6. In the same file, add said modal dialog:
```

Comment:
Where should this be added in the file?

Reply:
(In the example it seems it's added to the end of `<main>`

Reply:
Done